packages/docusaurus/src/webpack/utils.ts

```diff
@@ -206,88 +184,50 @@ export function applyConfigurePostCss(
     }
   }
 
+  // @ts-expect-error @types/webpack strikes again
   config.module?.rules.forEach(overridePostCssOptions);
 
   return config;
 }
 
-// See https://webpack.js.org/configuration/stats/#statswarningsfilter
-// @slorber: note sure why we have to re-implement this logic
-// just know that legacy had this only partially implemented, so completed it
-type WarningFilter = string | RegExp | ((warning: string) => boolean);
-function filterWarnings(
-  warningsFilter: WarningFilter[],
-  warnings: string[],
-): string[] {
-  function isWarningFiltered(warning: string): boolean {
-    return warningsFilter.some((warningFilter) => {
-      if (typeof warningFilter === 'string') {
-        return warning.includes(warningFilter);
-      } else if (warningFilter instanceof RegExp) {
-        return !!warning.match(warningFilter);
-      } else if (warningFilter instanceof Function) {
-        return warningFilter(warning);
-      } else {
-        throw new Error(`Unknown warningFilter type = ${typeof warningFilter}`);
-      }
-    });
-  }
-
-  return warnings.filter((warning) => !isWarningFiltered(warning));
-}
-
-export function compile(config: Configuration[]): Promise<Stats.ToJsonOutput> {
+export function compile(config: Configuration[]): Promise<void> {
   return new Promise((resolve, reject) => {
     const compiler = webpack(config);
     compiler.run((err, stats) => {
       if (err) {
-        reject(new Error(err.toString()));
+        reject(err);
       }
       // let plugins consume all the stats
       const allStats = stats?.toJson('errors-warnings');
       if (stats?.hasErrors()) {
-        allStats.errors.forEach((e) => {
-          console.error(e);
-        });
         reject(new Error('Failed to compile with errors.'));
       }
-      if (stats?.hasWarnings()) {
-        // Custom filtering warnings (see https://github.com/webpack/webpack/issues/7841).
-        let warnings = [...allStats.warnings];
-
-        const warningsFilter = ((config[0].stats as Stats.ToJsonOptionsObject)
-          ?.warningsFilter || []) as WarningFilter[];
-
-        if (Array.isArray(warningsFilter)) {
-          warnings = filterWarnings(warningsFilter, warnings);
-        }
-
-        warnings.forEach((warning) => {
+      if (allStats && stats?.hasWarnings()) {
+        allStats.warnings?.forEach((warning) => {
           console.warn(warning);
         });
       }
-      resolve(allStats);
+      resolve();
     });
   });
 }
 
 type AssetFolder = 'images' | 'files' | 'fonts' | 'medias';
 
-type FileLoaderUtils = {
-  loaders: {
-    file: (options: {folder: AssetFolder}) => Loader;
-    url: (options: {folder: AssetFolder}) => Loader;
-    inlineMarkdownImageFileLoader: string;
-    inlineMarkdownLinkFileLoader: string;
-  };
-  rules: {
-    images: () => RuleSetRule;
-    fonts: () => RuleSetRule;
-    media: () => RuleSetRule;
-    svg: () => RuleSetRule;
-    otherAssets: () => RuleSetRule;
-  };
-};
+interface Rules {
```

Comment:
why loaders and rules are now mixed? what about the inline file loaders?

Reply:
we have code using this:

```
const {
  loaders: {inlineMarkdownImageFileLoader},
} = getFileLoaderUtils();
```

 so it seems wrong to remove, and TS types do not match with runtime types